src/main.ts

```diff
@@ -1,22 +1,23 @@
 import * as core from '@actions/core';
-import { KindConfig, getKindConfig } from './kind';
+import { KindService } from './kind';
 import process from 'process';
+import * as go from './go';
 
 async function run() {
   try {
     checkEnvironment();
-    const cfg: KindConfig = getKindConfig();
-    const toolPath: string = await cfg.installKind();
+    const service: KindService = KindService.getInstance();
+    const toolPath: string = await service.installKind();
     core.addPath(toolPath);
-    await cfg.createCluster();
+    await service.createCluster();
   } catch (error) {
     core.setFailed((error as Error).message);
   }
 }
 
 function checkEnvironment() {
-  const supportedPlatforms: string[] = ['linux/x64'];
-  const platform = `${process.platform}/${process.arch}`;
+  const supportedPlatforms: string[] = ['linux/amd64'];
```

Comment:
It can be opened to linux/arm64 as [kind release note](https://github.com/kubernetes-sigs/kind/releases/tag/v0.11.1) is saying : 
> NOTE: these node images support amd64 and arm64 now. It remains possible to build custom images for other architectures (see the docs).

What do you think @radu-matei ?

Reply:
Oh this would be awesome!